asdf/commands/tests/test_edit.py

```diff
@@ -0,0 +1,231 @@
+                print("foo: 2", file=fd)  #  Change a value
+            else:
+                fd.write(l)
+
+    return oname
+
+
+def create_edit_larger(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value.  After this
+    line, add another line.  This will create an edited YAML file that will
+    have more characters than the YAML portion of the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_larger.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            fd.write(l)
+            if "foo" in l:
+                print("bar: 13", file=fd)  # Add a line
+
+    return oname
+
+
+def copy_base_asdf_equal(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with equal number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_equal.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+def copy_base_asdf_smaller(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with equal number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_smaller.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+def copy_base_asdf_larger(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with a larger number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_larger.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+@pytest.mark.parametrize("version", asdf.versioning.supported_versions)
+def test_edits(tmpdir,version):
+    #        Test:
+    # Create base ASDF file for testing
+    tmpdir = "/Users/kmacdonald/tmp"
+    asdf_base = create_base_asdf(tmpdir, version)
+
+    # Create base YAML file from base ASDF file
+    base, ext = os.path.splitext(asdf_base)
+    yaml_base = f"{base}.yaml"
+    # Run: asdftool edit -e -f {asdf_base} -o {yaml_base}
+    args = ["edit", "-e", "-f", f"{asdf_base}", "-o", f"{yaml_base}"]
+    main.main_from_args(args)
+
+    # Test smaller
+    # Create ASDF file to edit with larger sized YAML
+    asdf_smaller = copy_base_asdf_smaller(asdf_base)
+
+    # Create edited YAML file with larger number of characters
+    yaml_smaller = create_edit_smaller(yaml_base)
+
+    # Run: asdftool edit -s -f {yaml_larger} -o {asdf_larger}
+    args = ["edit", "-s", "-f", f"{yaml_smaller}", "-o", f"{asdf_smaller}"]
+    main.main_from_args(args)
+
+    af_smaller = asdf.open(asdf_smaller)
```

Comment:
Even though this is only a test, it's good to get in the habit of closing the AsdfFile after use, so that the underlying file pointer gets released.  The context manager makes this easy:

```python
with asdf.open(asdf_smaller) as af_smaller:
    assert af_smaller.tree["foo"] == 2
```

Reply:
I changed test checks to this format.